engine/evar_kinds.ml

```diff
@@ -35,3 +35,4 @@ type t =
   | MatchingVar of matching_var_kind
   | VarInstance of Id.t
   | SubEvar of subevar_kind option * Evar.t
+  | RecordFieldEvar of Constant.t * Names.constructor
```

Comment:
Maybe it would be easier to add "is this a missing record field" to QuestionMark.
~~~ocaml
(* maybe this should be a Projection.t *)
type is_missing_record_field = { field : Constant.t; record : Names.inductive }
...
  | QuestionMark of obligation_definition_status * Name.t * is_missing_record_field option
...
~~~

Reply:
Yes, I also think that in the current messy situation this is probably the easiest way.